cpp/src/arrow/filesystem/s3fs.cc

```diff
@@ -1433,11 +1455,12 @@ bool IsDirectory(std::string_view key, const S3Model::HeadObjectResult& result)
 class ObjectInputFile final : public io::RandomAccessFile {
  public:
   ObjectInputFile(std::shared_ptr<S3ClientHolder> holder, const io::IOContext& io_context,
-                  const S3Path& path, int64_t size = kNoSize)
+                  const S3Path& path, int64_t size = kNoSize, const std::string& c_key = "")
       : holder_(std::move(holder)),
         io_context_(io_context),
         path_(path),
-        content_length_(size) {}
+        content_length_(size),
+        sse_customer_key(c_key) {}
```

Comment:
Could you use `XXX_` style (`sse_customer_key_`) for a member variable name?

Reply:
ok, I would refine.